carla/models/catalog/catalog.py

```diff
@@ -234,3 +245,79 @@ def predict_proba(self, x):
             raise ValueError(
                 'Uncorrect backend value. Please use only "pytorch" or "tensorflow".'
             )
+
+    @property
+    def scaler(self):
+        """
+        Returns the fitted MinMax-scaler
+
+        Returns
+        -------
+        sklearn.MinMaxScaler()
+        """
+        return self._scaler
+
+    def set_scaler(self, data):
+        """
+        Sets and fits a new MinMax-scaler according to the data
+        Parameters
+        ----------
+        data : carla.data.Data()
+
+        Returns
+        -------
+        Fitted sklearn.MinMaxScaler()
+        """
+        return preprocessing.MinMaxScaler().fit(data.raw[self._continuous])
+
+    @property
+    def encoder(self):
+        """
+        Returns the fitted OneHotEncoder
+
+        Returns
+        -------
+        sklearn.OneHotEncoder
+        """
+        return self._encoder
+
+    def set_encoder(self, data):
+        """
+        Sets and fits a new OneHotEncoder according to data
+        Parameters
+        ----------
+        data : carla.data.Data()
+
+        Returns
+        -------
+        Fitted sklearn OneHotEncoder()
+        """
+        return preprocessing.OneHotEncoder(handle_unknown="ignore", sparse=False).fit(
+            data.raw[self._categoricals]
+        )
+
+    @property
+    def use_pipeline(self):
+        """
+        Returns if the ML model uses the pipeline for predictions
+
+        Returns
+        -------
+        bool
+        """
+        return self._use_pipeline
+
+    def set_use_pipeline(self, use_pipe):
```

Comment:
Lets make this a property setter instead aswell

```
@use_pipeline.setter
def use_pipeline()
```

Reply:
Nice, so with this declarator the interpreter automatically knows that this method is a setter?
I haven't worked with them to such an extend 😄

Reply:
We could also do this for set_scaler and set_encoder?